test/Scrutor.Tests/ScanningTests.cs

```diff
@@ -140,7 +175,7 @@ public void AutoRegisterAsMatchingInterface()
                     .AsMatchingInterface()
                     .WithTransientLifetime());
 
-            Assert.Equal(2, Collection.Count);
+            Assert.Equal(3, Collection.Count);

```

Comment:
Hm... cool this was what I planed to implement as next :). I have to look into it.
Just to make sure.

We have tons of code looking like this:
```cs
[ServiceDescriptor(typeof(ICoolService<Basic1, Basic2>))]
public class CoolServiceForBasic1Basic2 : ICoolService<Basic1, Basic2>{}

[ServiceDescriptor(typeof(ICoolService<Basic1, Basic3>))]
public class CoolServiceForBasic1Basic3 : ICoolService<Basic1, Basic3>{}

[ServiceDescriptor(typeof(ICoolService<Basic3, Basic4>))]
public class CoolServiceForBasic3Basic4 : ICoolService<Basic3, Basic4>{}

[ServiceDescriptor(typeof(ICoolService<Basic4, Basic5>))]
public class CoolServiceForBasic4Basic5 : ICoolService<Basic4, Basic5>{}
```

Where I would prefere something more like:
```cs
[ServiceDescriptor]
public class CoolServiceForBasic1Basic2 : ICoolService<Basic1, Basic2>{}

[ServiceDescriptor]
public class CoolServiceForBasic1Basic3 : ICoolService<Basic1, Basic3>{}

[ServiceDescriptor]
public class CoolServiceForBasic3Basic4 : ICoolService<Basic3, Basic4>{}

[ServiceDescriptor]
public class CoolServiceForBasic4Basic5 : ICoolService<Basic4, Basic5>{}
```
So I don't have to write duplicate code. I have changed the code in `AttributeSelector` the way that it would use the interface the class is implementing (if it is only one), and if not, then it would use the current implementation where it adds the current `type` as the `ServiceType` (https://github.com/khellang/Scrutor/blob/master/src/Scrutor/AttributeSelector.cs#L58).